Feat/schematic options

d-koppenhagen opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: closes #100


## What is the new behavior?

This PR will implement the options for schematics as discussed in #100 


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jorgeucano commented:
Hey, I’m return in 2 days to work ... and i will review all.. thanks for all of this work 💪

d-koppenhagen commented:
Awesome. I will also work on providing some schematic tests

aaronfrost commented on schematics/scully/src/add-post/index.ts:
```diff
@@ -1,27 +1,28 @@
-      context.logger.info(`✅️Blog ${name} file created`);
+`
+      );
+      context.logger.info(`✅️ Blog ${name} file created`);
```
Shouldn't this template include `${nameDasherized}` instead of `${name}`.

aaronfrost commented on schematics/scully/src/add-post/index.ts:
```diff
@@ -1,27 +1,28 @@
+    const targetDasherized = options.target ? strings.dasherize(options.target) : 'blog';
+    if (!host.exists(`./${targetDasherized}/${nameDasherized}.md`)) {
+      host.create(
+        `./blog/${nameDasherized}.md`,
```
This should probably be `./${targetDasherized}/${nameDasherized}.md`.

aaronfrost commented:
See my comments and then I can approve. @d-koppenhagen

d-koppenhagen commented:
@aaronfrost jep I will solve this later. Nonetheless a bit of docs and probably tests are missing. I will add them.

Also there is still an ongoing discussion in #100 about more options for posts for configuring meta data. I think your opinion to the suggestions I made would be great to have.

d-koppenhagen commented:
@jorgeucano i've also done a bit of refactoring and I added some more test specs. It would be good to check all this in detail and tests it again :)

jorgeucano commented:
hey @ d-koppenhagen this is a great job! Thanks, can you describe all the changes in a list? ... I need this to test the functionalities

d-koppenhagen commented:
Sure. I will do this later. 
But most things should be tested now by running `npm test` inside `schematics/scully` (I don’t know if the tests are also executed within a bigger test for scully in general using bazel?)
My plan was to add some more schematic tests today

jorgeucano commented:
I really love the idea, one of my next tasks is to start writing the test ... but, in this case, please add me the list and when we merge this ... we can create other pr ... I like a pr by functionality/problem/etc ... it is easier to test and work :)

d-koppenhagen commented:
Yes you are probably right. Let's merge this and face tests and more functionality in separate MR's and issues.

So here is the list:
- feature: provide more options for `@sculllyio/init:markdown` (#100)
  - `sourceDir`: define a source dir name (when not used, `name` is used instead)
  - `route`: define a route path before the `:slug` (when not used, `name` is used instead)
- feature: provide more options for `@sculllyio/init:post` (#100)
  - `target`: define the target directory for the new post file (by default `blog` is used)
  - `metaDataFile`: use a meta data yaml template from a file for usage within the post (`title` will be overridden by the `name` option if set)
- feature: use `yyyy-mm-dd` as date format in the file name for the initially created markdown file when using `@sculllyio/init:markdown` (31e1ebdd7ad990de14efb7155ad7214dcf54ce6a)
  - just as it is more common
- tests: added `test-utils.ts` for easier test setup (01f4531ea24013e3efe4503632ad0d1a33f0f80d)
- docs: updated docs file with schematics options for `@sculllyio/init:markdown` and `@sculllyio/init:post` (#100)
- refactoring: split `ng-add` schematic into smaller chainable rules (50e7bec04e3ff4a3abcfa414b205f25495a571a6)
- tests: add tests for several schematics (`ng-add`, `scully`, `create-markdown`, `add-post`)
- refactor: add schema for `add-blog` schematic (68c5c0cbd6cadb79bb31b96b495b4f5b8da8d138)

jorgeucano commented:
@d-koppenhagen i created a little change for support nx with schematics, can you pull from master and merge :) after this i will Approve :) thanks for all of this work

d-koppenhagen commented:
@jorgeucano already done 😃 see 4fd2cb0

jorgeucano commented:
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉
